sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample11_MockingClientTypes.md

```diff
@@ -0,0 +1,456 @@
+
+# Mocking Client Types
+
+Event Hubs is built to support unit testing with mocks, as described in the [Azure SDK Design Guidelines](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-mocking). This is an important feature of the library that allows developers to write tests that are completely focused on their own application logic, though they depend on the Event Hubs types.
+
+The following examples focus on scenarios likely to occur in applications, and demonstrate how to mock the Event Hubs types typically used in each scenario. The code snippets utilize the mock object framework, Moq, in order to provide practical examples. However, many mocking frameworks exist and can be used with the same approach in mind.
+
+## Reading Event Hub metadata with an `EventHubProducerClient` or `EventHubBufferedProducerClient`
+
+Many applications make decisions for publishing based on the properties of the Event Hub itself or the properties of its partitions. Both can be mocked using the `EventHubsModelFactory`. The following snippet demonstrates how to mock Event Hub properties for both the `EventHubBufferedProducerClient` and the `EventHubProducerClient`, and how to set up the mocked producer clients to return a cohesive set of values for each of the accompanying getter methods.
+
+These mocked properties can be used to test expected application behavior given a specific set of Event Hub properties. These property types can be mocked in the same way for the `EventHubConsumerClient`.
+
+```C# Snippet:EventHubs_Sample11_MockingProducerProperties
```

Comment:
Looking at this comment, I agree!  I'd suggest putting one Mocking Properties section after the Producer/BufferedProducer,Consumer sections.    Leave the PartitionReceiver for last.